argutils/viz.py

```diff
@@ -30,25 +34,28 @@ def draw(ts, ax, use_ranked_times=None, tweak_x=None, arrows=False):
         except (TypeError, KeyError):
             labels[nd.id] = str(nd.id)
 
-    pos = nx_get_dot_pos(G)
+    positions = nx_get_dot_pos(G)
```

Comment:
This would be a lot simplier/easier to follow if it was either/or. Why do you want to allow updating pos dictionary?

Reply:
I assumed that we might want to only change some of the node positions (although I guess normally we would change all of them). It seemed easy enough to simply replace all the existing positions, but actually we could do that with positions.update(pos) rather than an explicit loop.

Reply:
But we're not *actually* using it, right? These PRs would be a lot quicker if they included only functionality that we need. 

I know this is annoying, but it's far easier for me or anyone else to understand these functions if they just have the functionality that's being used. Adding on loads of bells and whistles just because we can and we might want them later makes it much harder to follow. 

It also makes them much harder to change. Suppose I wanted to add some functionality that conflicted with this - how would I know that this is a "we might want this later" feature, which can be dropped vs something that's actively being used?

Reply:
If we do want it later, it's a 30 second job to do it. It's not like we've written test cases to cover the functionality or anything, so it really is just a waste of time putting in extra stuff before it's needed.

Reply:
Sure: this just seemed like the most obvious way to do it when it occurred to me.

Reply:
Done